File: java-end/src/strategyconsoleapplication/Product.java
package strategyconsoleapplication;

public class Product {
    private String name;
    private String recommendedBy;

    public Product(String name, String recommendedBy) {
        this.name = name;
        this.recommendedBy = recommendedBy;
    }

    public String getName() {
        return name;
    }

    public String getRecommendedBy() {
        return recommendedBy;
    }

    @Override
    public String toString() {
        return name + ", " + recommendedBy;
    }
}

File: begin/StrategyConsoleApplication/StrategyConsoleApplication/Product.cs
namespace StrategyConsoleApplication
{
    public class Product
    {
        public string Name { get; private set; }
        public string RecommendedBy { get; private set; }

        public Product(string name, string recommendedBy)
        {
            Name = name;
            RecommendedBy = recommendedBy;
        }

        public override string ToString()
        {
            return string.Format("{0}", Name);
        }
    }
}